utils/_context/_scenarios.py

```diff
@@ -837,7 +846,7 @@ class scenarios:
 
     otel_tracing_e2e = OpenTelemetryScenario(
         "OTEL_TRACING_E2E",
-        weblog_env={"DD_API_KEY": os.environ.get("DD_API_KEY"), "DD_SITE": os.environ.get("DD_SITE"),},
+        weblog_env={"DD_API_KEY": os.environ.get("DD_API_KEY_2"), "DD_SITE": os.environ.get("DD_SITE", "datad0g.com"),},
```

Comment:
At this point, `DD_API_KEY_2` may not be in environ. `configure` is the function executed whem your scenario is actually the executed scenatio. it must be handled inside this function with : 

```
    def configure(self):
        super().configure()
        self.weblog.environment["key"] = os.environ...
```

Reply:
Sounds good, moved all env setup to `configure`